src/RemoteAPI/related/class-related-caching-decorator.php

```diff
@@ -42,10 +41,10 @@ class Related_Caching_Decorator implements Proxy {
 	/**
 	 * Constructor.
 	 *
-	 * @param Proxy           $proxy The Proxy object to cache.
-	 * @param WP_Object_Cache $cache The WordPress object cache instance.
+	 * @param Proxy         $proxy The Proxy object to cache.
+	 * @param Cache_Adapter $cache An object cache instance.
 	 */
-	public function __construct( Proxy $proxy, WP_Object_Cache $cache ) {
+	public function __construct( Proxy $proxy, Cache_Adapter $cache ) {
 		$this->decorated_proxy = $proxy;
```

Comment:
Since it's _this_ class that is doing the decorating, then this property could just be named as `$proxy`, rather than `$decorated_proxy`.